src/back/msl.rs

```diff
@@ -546,6 +546,19 @@ impl<W: Write> Writer<W> {
                             &crate::TypeInner::Vector { size, kind, width },
                             &crate::TypeInner::Vector { .. },
                         ) => MaybeOwned::Owned(crate::TypeInner::Vector { size, kind, width }),
+                        (
```

Comment:
I don't understand this matching arm. Is it possible to have the following expression in Metal?

Reply:
Ah yes I think I already answered my own question. I have search through the Metal documentation and it should be possible: "A right vector operand is treated as a column vector and a left vector operand as a row vector.".

The left operand as vector is not yet implemented, are there any plans for this?

Reply:
I was implemented as much as needed to get the SPV examples running (more concretely, the cube shaders here, and the wgpu-rs shaders). We can consider expanding this in other PRs. Honestly, I don't know if IR needs to care about "vector * matrix" syntax... since it's trivial to polyfill as `translate(matrix * vector)`. Let's see where the working group goes with this and follow-up?

Reply:
Sure no problem, I was just wondering, never made a Metal shader :)